tests/unit/test_table.py

```diff
@@ -2139,6 +2150,49 @@ def test_to_arrow_max_results_w_create_bqstorage_warning(self):
             and "REST" in str(warning)
         ]
         self.assertEqual(len(matches), 1, msg="User warning was not emitted.")
+        self.assertIn(
+            __file__, str(matches[0]), msg="Warning emitted with incorrect stacklevel"
+        )
+        mock_client._ensure_bqstorage_client.assert_not_called()
+
+    @unittest.skipIf(pyarrow is None, "Requires `pyarrow`")
+    @unittest.skipIf(
+        bigquery_storage is None, "Requires `google-cloud-bigquery-storage`"
+    )
```

Comment:
This might not cause a merge conflict in the `v3` branch, but still represents a "semantic conflict". We made `pyarrow` and `bigquery-storage` required there, thus these `skipIf` decorators are unnecessary.

Since unwanted changes can silently creep in into the `v3` branch, I suggest periodically opening PRs against `v3` that bring in the changes from `master`, which can we then review before merging.